tests/test_single_topology_confgen.py

```diff
@@ -0,0 +1,100 @@
+        mol_ab_frame = pdb_writer.convert_single_topology_mols(ligand_frame, atom_map_mixin)
+        writer.write_frame(np.concatenate([host_frame, mol_ab_frame]) * 10)
+    writer.close()
+
+
+def run_edge(mol_a, mol_b, protein_path):
+
+    threshold = 2.0
+    core, smarts = atom_mapping.get_core_with_alignment(mol_a, mol_b, threshold=threshold)
+    res = utils.plot_atom_mapping_grid(mol_a, mol_b, smarts, core)
+    with open(f"edge_map_{get_mol_name(mol_a)}_{get_mol_name(mol_b)}.svg", "w") as fh:
+        fh.write(res)
+
+    ff = Forcefield.load_from_file(DEFAULT_FF)
+    st = SingleTopology(mol_a, mol_b, core, ff)
+
+    lambda_schedule = np.linspace(0, 1, 12)
+    seed = 2023
+
+    # solvent
+    box_width = 4.0
+    solvent_sys, solvent_conf, solvent_box, solvent_top = builders.build_water_system(box_width, ff.water_ff)
+    solvent_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    solvent_host_config = HostConfig(solvent_sys, solvent_conf, solvent_box)
+    initial_states = setup_initial_states(st, solvent_host_config, DEFAULT_TEMP, lambda_schedule, seed)
+    all_frames = [state.x0 for state in initial_states]
+    write_trajectory_as_pdb(
+        mol_a,
+        mol_b,
+        core,
+        all_frames,
+        solvent_top,
+        f"solvent_{get_mol_name(mol_a)}_{get_mol_name(mol_b)}.pdb",
+    )
+
+    # complex
+    complex_sys, complex_conf, _, _, complex_box, complex_top = builders.build_protein_system(
+        protein_path, ff.protein_ff, ff.water_ff
+    )
+    complex_box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+    complex_host_config = HostConfig(complex_sys, complex_conf, complex_box)
+    initial_states = setup_initial_states(st, complex_host_config, DEFAULT_TEMP, lambda_schedule, seed)
+    all_frames = [state.x0 for state in initial_states]
+    write_trajectory_as_pdb(
+        mol_a,
+        mol_b,
+        core,
+        all_frames,
+        complex_top,
+        f"complex_{get_mol_name(mol_a)}_{get_mol_name(mol_b)}.pdb",
+    )
+
+
+@pytest.mark.nightly(reason="Takes a while to run")
+def test_confgen():
+
+    protein_path = "timemachine/testsystems/data/hif2a_nowater_min.pdb"
+    ligands = "timemachine/datasets/fep_benchmark/hif2a/ligands.sdf"
+    mols = [mol for mol in Chem.SDMolSupplier(ligands, removeHs=False)]
+
+    results_csv = "timemachine/datasets/fep_benchmark/hif2a/results_edges_5ns.csv"
+    with open(results_csv) as csvfile:
+        reader = csv.reader(csvfile, delimiter=",")
+        next(reader)
+        rows = [row for row in reader]
+        for row_idx, row in enumerate(rows):
+            mol_a_name, mol_b_name, exp_ddg, fep_ddg, fep_ddg_err, ccc_ddg, ccc_ddg_err = row
+            print("Processing", mol_a_name, "->", mol_b_name)
+            mol_a = get_mol_by_name(mols, mol_a_name)
+            mol_b = get_mol_by_name(mols, mol_b_name)
+
+            run_edge(mol_a, mol_b, protein_path)
```

Comment:
nit: testing: should this PR introduce a unit test of `optimize_coordinates`?

(In addition to the affected integration tests that rely on `setup_initial_states`...)

Reply:
I can cherry pick a couple of edges.